### ISSUE TITLE
Internal compile error when accessing property of return value inside a condition with language out ES2017

### ISSUE THREAD

**Issue by WearyMonkey** [google/closure-compiler#3658]
Affects: v20200719

```
const b = () => ({ x: '' })

function main() {
    let a;
    if (Math.random()) {
        a = b();
        console.log(a.x);
    }

}

main();
```

```
java -jar compiler.jar --language_out ECMASCRIPT_2017 -O ADVANCED --js a.js
java.lang.RuntimeException: INTERNAL COMPILER ERROR.
Please report this problem.

ROOT : global this
  Node(IF): a.js:5:4
    if (Math.random()) {
  Parent(BLOCK): a.js:3:16
function main() {

	at com.google.common.base.Preconditions.checkState(Preconditions.java:507)
	at com.google.javascript.jscomp.InlineObjectLiterals$InliningBehavior.splitObject(InlineObjectLiterals.java:439)
	at com.google.javascript.jscomp.InlineObjectLiterals$InliningBehavior.afterExitScope(InlineObjectLiterals.java:100)
	at com.google.javascript.jscomp.ReferenceCollectingCallback.exitScope(ReferenceCollectingCallback.java:262)
	at com.google.javascript.jscomp.NodeTraversal.popScope(NodeTraversal.java:1096)
	at com.google.javascript.jscomp.NodeTraversal.popScope(NodeTraversal.java:1087)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:903)
	at com.google.javascript.jscomp.NodeTraversal.traverseChildren(NodeTraversal.java:1037)
	at com.google.javascript.jscomp.NodeTraversal.handleScript(NodeTraversal.java:842)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:877)
	at com.google.javascript.jscomp.NodeTraversal.traverseChildren(NodeTraversal.java:1037)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:905)
	at com.google.javascript.jscomp.NodeTraversal.traverseRoots(NodeTraversal.java:445)
	at com.google.javascript.jscomp.ReferenceCollectingCallback.process(ReferenceCollectingCallback.java:111)
	at com.google.javascript.jscomp.InlineObjectLiterals.process(InlineObjectLiterals.java:66)
	at com.google.javascript.jscomp.PhaseOptimizer$NamedPass.process(PhaseOptimizer.java:317)
	at com.google.javascript.jscomp.PhaseOptimizer$Loop.process(PhaseOptimizer.java:462)
	at com.google.javascript.jscomp.PhaseOptimizer.process(PhaseOptimizer.java:232)
	at com.google.javascript.jscomp.Compiler.performOptimizations(Compiler.java:2434)
	at com.google.javascript.jscomp.Compiler.lambda$stage2Passes$1(Compiler.java:816)
	at com.google.javascript.jscomp.CompilerExecutor$2.call(CompilerExecutor.java:102)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1135)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:844)
Caused by: java.lang.IllegalStateException: ROOT : global this
	... 25 more
```

**Comment by WearyMonkey** [comment]
This error is still being thrown in `20200920.0.0-nightly`

**Comment by WearyMonkey** [comment]
@brad4d this one next? :)

**Comment by WearyMonkey** [comment]
Output with `--print_source_after_each_pass`

```
// parseInputs yields:
// ************************************
'use strict';const b=()=>({x:""});function main(){let a;if(Math.random()){a=b();console.log(a.x)}}main();
// ************************************

// normalize yields:
// ************************************
'use strict';const b=()=>{return{x:""}};function main(){let a$jscomp$1;if(Math.random()){a$jscomp$1=b();console.log(a$jscomp$1.x)}}main();
// ************************************

// earlyInlineVariables yields:
// ************************************
'use strict';const b=()=>{return{x:""}};(function main(){let a$jscomp$1;if(Math.random()){a$jscomp$1=b();console.log(a$jscomp$1.x)}})();
// ************************************

// removeUnusedCode yields:
// ************************************
'use strict';const b=()=>{return{x:""}};(function(){let a$jscomp$1;if(Math.random()){a$jscomp$1=b();console.log(a$jscomp$1.x)}})();
// ************************************

// inlineFunctions yields:
// ************************************
'use strict';const b=()=>{return{x:""}};{let a$jscomp$inline_0;if(Math.random()){a$jscomp$inline_0=b();console.log(a$jscomp$inline_0.x)}};
// ************************************

// inlineFunctions yields:
// ************************************
'use strict';{let a$jscomp$inline_0;if(Math.random()){a$jscomp$inline_0={x:""};console.log(a$jscomp$inline_0.x)}};
// ************************************
java.lang.RuntimeException: INTERNAL COMPILER ERROR.
Please report this problem.

ROOT : global this
  Node(IF): a.js:5:4
    if (Math.random()) {
  Parent(BLOCK): a.js:3:16
function main() {

	at com.google.common.base.Preconditions.checkState(Preconditions.java:508)
	at com.google.javascript.jscomp.InlineObjectLiterals$InliningBehavior.splitObject(InlineObjectLiterals.java:439)
	at com.google.javascript.jscomp.InlineObjectLiterals$InliningBehavior.afterExitScope(InlineObjectLiterals.java:100)
	at com.google.javascript.jscomp.ReferenceCollectingCallback.exitScope(ReferenceCollectingCallback.java:262)
	at com.google.javascript.jscomp.NodeTraversal.popScope(NodeTraversal.java:1096)
	at com.google.javascript.jscomp.NodeTraversal.popScope(NodeTraversal.java:1087)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:903)
	at com.google.javascript.jscomp.NodeTraversal.traverseChildren(NodeTraversal.java:1037)
	at com.google.javascript.jscomp.NodeTraversal.handleScript(NodeTraversal.java:842)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:877)
	at com.google.javascript.jscomp.NodeTraversal.traverseChildren(NodeTraversal.java:1037)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:905)
	at com.google.javascript.jscomp.NodeTraversal.traverseRoots(NodeTraversal.java:445)
	at com.google.javascript.jscomp.ReferenceCollectingCallback.process(ReferenceCollectingCallback.java:111)
	at com.google.javascript.jscomp.InlineObjectLiterals.process(InlineObjectLiterals.java:66)
	at com.google.javascript.jscomp.PhaseOptimizer$NamedPass.process(PhaseOptimizer.java:317)
	at com.google.javascript.jscomp.PhaseOptimizer$Loop.process(PhaseOptimizer.java:462)
	at com.google.javascript.jscomp.PhaseOptimizer.process(PhaseOptimizer.java:232)
	at com.google.javascript.jscomp.Compiler.performOptimizations(Compiler.java:2437)
	at com.google.javascript.jscomp.Compiler.lambda$stage2Passes$1(Compiler.java:817)
	at com.google.javascript.jscomp.CompilerExecutor$2.call(CompilerExecutor.java:102)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1135)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:844)
Caused by: java.lang.IllegalStateException: ROOT : global this
	... 25 more
```

**Comment by rishipal** [comment]
This happens not just in conditional, and not just on property access of a return value, but also on a simple, unconditional property access within a BLOCK scope. Example:

```
{let a; a={x:""};console.log(a.x);}
```
crashes
```
 SCRIPT 1 [source_file: externs] [input_id: InputId: externs] [feature_set: []]
  Node(EXPR_RESULT): testcode:1:17
{let a; a={x:""};console.log(a.x);}
  Parent(BLOCK): testcode:1:0
{let a; a={x:""};console.log(a.x);}

java.lang.RuntimeException: 
	at com.google.common.base.Preconditions.checkState(Preconditions.java:508)
	at com.google.javascript.jscomp.InlineObjectLiterals$InliningBehavior.splitObject(InlineObjectLiterals.java:442)
	at com.google.javascript.jscomp.InlineObjectLiterals$InliningBehavior.afterExitScope(InlineObjectLiterals.java:100)
	at com.google.javascript.jscomp.ReferenceCollectingCallback.exitScope(ReferenceCollectingCallback.java:262)
        ...
	at com.google.javascript.jscomp.InlineObjectLiteralsTest.repro2(InlineObjectLiteralsTest.java:135)
	....
```

**Comment by rishipal** [comment]
Fixed with da0e34af0a929f02a8698946ad260ae34e860f8b
